pnet_datalink/src/pcap.rs

```diff
@@ -11,32 +11,41 @@ use std::path::Path;
 
 use self::pcap::{Active, Activated};
 
-use datalink::{DataLinkReceiver, DataLinkSender, NetworkInterface};
-use datalink::Channel::Ethernet;
+use {DataLinkReceiver, DataLinkSender, NetworkInterface};
+use Channel::Ethernet;
 
 /// Configuration for the pcap datalink backend
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub struct Config {
-    /// The size of buffer to use when reading packets. Defaults to 4096
+    /// The size of buffer to use when reading packets. Must be at least
+    /// 65516 with pcap.
     pub read_buffer_size: usize,
 
     /// The read timeout. Defaults to None.
     pub read_timeout: Option<Duration>,
 }
 
-impl<'a> From<&'a datalink::Config> for Config {
-    fn from(config: &datalink::Config) -> Config {
-        Config{
+impl<'a> From<&'a super::Config> for Config {
+    fn from(config: &super::Config) -> Config {
+        let mut c = Config{
             read_buffer_size: config.read_buffer_size,
             read_timeout: config.read_timeout,
+        };
+        // pcap is unique in that the buffer size must be greater or equal to
+        // MAXIMUM_SNAPLEN, which is currently hard-coded to 65536
+        // So, just reset it to the default.
+        if c.read_buffer_size < 65536 {
+            c.read_buffer_size = Config::default().read_buffer_size;
         }
+        c
     }
 }
 
 impl Default for Config {
     fn default() -> Config {
         Config{
-            read_buffer_size: 4096,
+            // This is the libpcap default
+            read_buffer_size: 2097152,
```

Comment:
Where did you get this number from? I read here: https://docs.rs/pcap/0.7.0/pcap/struct.Capture.html#method.buffer_size And it says the default is `1000000`. But I have no idea if those are related. I just looked it up because I think it's good if we can get as many numbers as possible as variables from our dependency rather than hardcoding them.

Reply:
My source is this - https://github.com/the-tcpdump-group/libpcap/blob/master/pcap-linux.c#L3758. I wonder if libpcap supports sending a "0" buffer size for all OSs.

Reply:
If it does. And/or if their way of activating their default value is to send zero, then I think we should send zero.

If not. I think the `2 * 1024 * 1024` is much clearer than the computed value. It makes it easy to see that it is indeed 2 MiB.